### PR TITLE
Zihan/wd 65 etcd jar

### PR DESCRIPTION
**Pull request by bonellon**
Added non-runnable ETCD jar to perform CRUD operations on directories & keys

### CODE REVIEW

File: Code/ETCDLibrary/build.gradle [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    dependencies {
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:2.0.5.RELEASE")
+    }
+}
+
+apply plugin: 'java'
+apply plugin: 'eclipse'
+apply plugin: 'idea'
+apply plugin: 'org.springframework.boot'
+apply plugin: 'io.spring.dependency-management'
```
Comment (ats0stv) [inline Code/ETCDLibrary/build.gradle]: Please apply the sonarqube plugin and run it via sonarqube

File: Code/ETCDLibrary/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ *
+ * The settings file is used to specify which projects to include in your build.
+ * 
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user guide at https://docs.gradle.org/4.8.1/userguide/multi_project_builds.html
+ */
+
+rootProject.name = 'ETCDLibrary'
+include "REST"
+project(":REST").projectDir = new File("../Libraries/RESTLibrary");
```
Comment (ats0stv) [inline Code/ETCDLibrary/settings.gradle]: Please include new line at EOF

File: Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import java.io.IOException;
+import java.util.Optional;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+
+public class ETCDLibrary {
+
+	private static String baseUrl = "http://35.246.14.12:2379/v2/keys/";
```
Comment (ats0stv) [inline Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java]: Please get this info from the configuration file and do not hard-code

File: Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		
+	public static String ReadDirectory(String directoryName) throws IOException {
+		
+		String url = (baseUrl + directoryName).trim();
+		HttpResponse response = Library.GET(url, Optional.empty());
+		
+		 HttpEntity responseEntity = response.getEntity();
+	     String responseXml = EntityUtils.toString(responseEntity);
```
Comment (ats0stv) [inline Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java]: Is it `responseXml`. AFAIK, we get a JSON. And why should we parse it as a `String`?

File: Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	public static String DeleteDirectory(String directoryName) throws IOException {
+		
+		String url = baseUrl + directoryName+"?dir=true";
+		
+		HttpResponse response = Library.DELETE(url, Optional.empty());
+		
+		HttpEntity responseEntity = response.getEntity();
+	    String responseXml = EntityUtils.toString(responseEntity);
+	     
+	    return responseXml;
```
Comment (ats0stv) [inline Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java]: Same as above and for below one. Why `responseXml`?

File: Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		HttpResponse response = Library.GET(url, Optional.empty());
+		
+		 HttpEntity responseEntity = response.getEntity();
+	     String responseXml = EntityUtils.toString(responseEntity);
+	     
+	     return responseXml;
+		
+	}
+	
+	public static String DeleteDirectory(String directoryName) throws IOException {
```
Comment (ats0stv) [inline Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java]: Is there a reason for these methods top and below to be static? Please explain.

File: Code/Libraries/RESTLibrary/build.gradle [hunk 7]
```diff
@@ -13,12 +13,10 @@ apply plugin: 'idea'
 apply plugin: 'org.springframework.boot'
 apply plugin: 'io.spring.dependency-management'
 
-
 jar { enabled = true }
```
Comment (ats0stv) [inline Code/Libraries/RESTLibrary/build.gradle:16]: I don't think these are necessary

File: Code/Libraries/RESTLibrary/src/main/java/ie/tcd/wayfinders/restLibrary/Library.java [hunk 8]
```diff
@@ -56,11 +56,9 @@ private static HttpResponse handleRest(HttpRequestBase requestMethod, Optional<M
...
         }
-
+       
         HttpClient client = HttpClientBuilder.create().build();
-        
         HttpResponse response = client.execute(requestMethod);
-        
         return response;
     }
 }
```
Comment (ats0stv) [inline Code/Libraries/RESTLibrary/src/main/java/ie/tcd/wayfinders/restLibrary/Library.java]: New line at EOF

**Review by ats0stv — CHANGES_REQUESTED** [review]
Please remove the /bin/* files

File: Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java (hunk 6 above)
Comment (bonellon) [reply Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java]: removed

File: Code/Libraries/RESTLibrary/build.gradle (hunk 7 above)
Comment (bonellon) [reply Code/Libraries/RESTLibrary/build.gradle:16]: we need restLibrary to be compiled as jar so that other gradle projects can import it no?

**Pull request closed without merging**